design_drafts/flexible_joint_states_msg.md

```diff
@@ -34,28 +56,122 @@ float64[] velocity
 float64[] effort
 ```
 
-## First proposal
+The following proposal reflects a map-like (similar to what `std::unordered_map` represents in C++ or a hash table in other languages) datatype where not only three statically chosen value-identifier are attached to a joint name, but can be set dynamically.
+
+### Proposal for a Dynamic Joint State Message
 
 ```
 Header header
 
-string[] name
+string[] interface_name
+InterfaceValue[] interface_value
```

Comment:
We should iterate a bit on these names, they are way too general for my taste.
Please feel free to resolve this once read, don't want to hold up the PR.